proto/aserto/directory/model/v3/model.proto

```diff
@@ -7,43 +7,50 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/model/v
 import "google/api/field_behavior.proto";
 import "google/protobuf/empty.proto";
 
-import "aserto/directory/common/v3/common.proto";
-
 service Model {
-    rpc GetManifest(GetManifestRequest) returns (stream GetManifestResponse) {};
-    rpc SetManifest(stream SetManifestRequest) returns (SetManifestResponse) {};
+    rpc GetModel(GetModelRequest) returns (stream GetModelResponse) {};
+    rpc SetModel(stream SetModelRequest) returns (SetModelResponse) {};
 }
 
-message GetManifestRequest {
+message Metadata {
     // manifest name (unique, lc-string)
-    string name                                                 = 1     [(google.api.field_behavior) = REQUIRED];
+    string  name                      = 1;
     // manifest version (semver notation)
-    string version                                              = 2     [(google.api.field_behavior) = REQUIRED];
+    string  version                   = 2;
 }
 
-message GetManifestResponse {
-    // manifest instance
-    aserto.directory.common.v3.Manifest result                  = 1;
+enum OnDelete {
+    ON_DELETE_UNKNOWN                  = 0x0;  // default, will be treated as fail
```

Comment:
The unset state should fail with an undefined argument error not double for the next one, `will be treated as fail`